### PR TITLE
Update HoverHandler for more common usage

### PR DESCRIPTION
**Pull request by vdubr**

### CODE REVIEW

File: src/components/HoverHandler/index.js [hunk 1]
```diff
@@ -40,92 +77,92 @@ const HoverHandler = ({
...
 				y: state.y,
 			}}
 		>
-			{children}
-			{state.popupContent || state.data ? renderPopup() : null}
-		</HoverContext.Provider>
-	);
+			<div ref={ref} style={{height: '100%', width: '100%'}}>
+				{children}
+				{/* {popup ? this.renderPopup() : null} */}
```
Comment (vlach1989) [inline src/components/HoverHandler/index.js:161]:
```suggestion

```

File: src/components/HoverHandler/position.js [hunk 2]
```diff
@@ -60,13 +60,14 @@ const getTopPossition = (
 
 	// return [posX, posY]
 	return {
-		position: {posX: posX - BBox[3], posY: posY},
+		// position: {posX: posX - BBox[3], posY: posY},
```
Comment (vlach1989) [inline src/components/HoverHandler/position.js:63]:
```suggestion

```

File: src/components/HoverHandler/position.js [hunk 3]
```diff
@@ -114,13 +115,14 @@ const getBottomPossition = (
 	];
 
 	return {
-		position: {posX: posX - BBox[3], posY: posY},
+		// position: {posX: posX - BBox[3], posY: posY},
```
Comment (vlach1989) [inline src/components/HoverHandler/position.js:118]:
```suggestion

```

File: src/components/HoverHandler/position.js [hunk 4]
```diff
@@ -164,13 +166,14 @@ const getRightPossition = (
 	];
 
 	return {
-		position: {posX: posXAbsoplute - BBox[3], posY: posY - BBox[0]},
+		// position: {posX: posXAbsoplute - BBox[3], posY: posY - BBox[0]},
```
Comment (vlach1989) [inline src/components/HoverHandler/position.js:169]:
```suggestion

```

File: src/components/HoverHandler/position.js [hunk 5]
```diff
@@ -214,7 +217,8 @@ const getLeftPossition = (
 	];
 
 	return {
-		position: {posX: posXAbsoplute - BBox[3], posY: posY - BBox[0]}, //relative to element
+		// position: {posX: posXAbsoplute, posY: posY - BBox[0]}, //relative to element
```
Comment (vlach1989) [inline src/components/HoverHandler/position.js:220]:
```suggestion

```

File: src/components/HoverHandler/position.js [hunk 6]
```diff
@@ -227,104 +231,99 @@ const checkBoundatiesConflict = (innerBoundingRect, padding, BBox) => {
...
+	let conflict = true;
+	let pos = null;
+	switch (position) {
+		case 'top': {
+			const topPos = getTopPosition(
+				origPosX,
+				origPosY,
+				width,
+				height,
+				hoveredElemen,
```
Comment (vlach1989) [inline src/components/HoverHandler/position.js:254]: Fix typo multiple times

File: src/components/HoverHandler/Popup/index.js [hunk 7]
```diff
@@ -71,45 +81,50 @@ const Popup = ({
...
-	const style = getStyleSelf();
+	const style = getPopupStyle();
 
 	let classes = classnames('ptr-popup', {
 		compressed: compressed,
 	});
 
 	if (ref && !ref.current) {
+		// FIXME - test
+		// commented for world water use in chart
```
Comment (vlach1989) [inline src/components/HoverHandler/Popup/index.js:109]:
```suggestion

```

**Review by vlach1989 — CHANGES_REQUESTED** [review]

File: src/components/HoverHandler/position.js [hunk 8]
```diff
@@ -227,104 +231,99 @@ const checkBoundatiesConflict = (innerBoundingRect, padding, BBox) => {
...
-			}
-			default:
-				break;
+const getPosition = (
+	position,
+	origPosX,
+	origPosY,
+	width,
+	height,
+	hoveredElemen,
```
Comment (vlach1989) [inline src/components/HoverHandler/position.js:240]: Fix typo multiple times

**Review by vlach1989 — CHANGES_REQUESTED** [review]

File: src/components/HoverHandler/position.js (hunk 6 above)
Comment (vdubr) [reply src/components/HoverHandler/position.js]: fixed: d0992f5beb5c3bfb780e65dcf7a0105a602e1e55

File: src/components/HoverHandler/position.js (hunk 8 above)
Comment (vdubr) [reply src/components/HoverHandler/position.js]: fixed: d0992f5beb5c3bfb780e65dcf7a0105a602e1e55

**Review by vlach1989 — APPROVED** [review]

**Pull request merged**